src/metabase/util/schema.clj

```diff
@@ -198,58 +212,108 @@
 ;;; |                                                 USEFUL SCHEMAS                                                 |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(spec/def ::defschema-args
+  (spec/cat
+   :name   symbol?
+   :docstr (spec/? string?)
+   :body   (spec/* any?)))
+
+(defmacro defschema
+  "Define a Malli schema"
+  [& args]
+  (let [{:keys [name docstr body]} (spec/conform ::defschema-args args)]
+    `(def ~(vary-meta name assoc :doc docstr) (mc/schema ~@body))))
+
```

Comment:
I think this schema-mirrored malli api should go into `metabase.util.malli`. that would make it easier to get all the malli things in the same place

Also, malli's parse(r) is like spec's conform(er). Let's not add spec or p.schema things if we can avoid it

```clojure
(def ^{:private true :arglists '([[name ?docstring & body]])} defschema-arg-parser
  (mc/parser [:catn
              [:name symbol?]
              [:doc-string [:? string?]]
              [:body [:* any?]]]))

(defmacro defschema
  "Define a Malli schema"
  [& args]
  (let [{:keys [name docstr body]} (defschema-arg-parser args)]
    `(def ~(vary-meta name assoc :doc docstr) (mc/schema ~@body))))
```